msgpack_entity.go

```diff
@@ -0,0 +1,52 @@
+package restful
+
+import (
+	"io"
+	"io/ioutil"
+
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+// NewEntityAccessorMPack returns a new EntityReaderWriter for accessing MessagePack content.
+// This package is not initialized with such an accessor using the MIME_MSGPACK contentType.
+func NewEntityAccessorMsgPack(contentType string) EntityReaderWriter {
+	return entityMsgPackAccess{ContentType: contentType}
+}
+
+// entityOctetAccess is a EntityReaderWriter for Octet encoding
+type entityMsgPackAccess struct {
+	// This is used for setting the Content-Type header when writing
+	ContentType string
+}
+
+// Read unmarshalls the value from byte slice and using msgpack to unmarshal
+func (e entityMsgPackAccess) Read(req *Request, v interface{}) error {
+	data, err := ioutil.ReadAll(req.Request.Body)
```

Comment:
I would use the NewDecoder function directly on the Request.Body instead.
That way you do not have to read all the bytes first. It will be more effictient that way.